### ISSUE TITLE
Not possible to add KeeShared file the second time after deletion

### ISSUE THREAD

**Issue by atiris** [keepassxreboot/keepassxc#6012]
## Overview

After delete KeeShare Group all items from this KeeShare file are invisible after next settings KeeShare to the same file.

## Steps to Reproduce

1. I created two kdbx databases with this structure:

local.kdbx

    Root LOCAL
    - Group TEST (group)

shared.kdbx

    Root SHARED
    - Pass SHARED (entry)
    - Group SHARED (group)
      - Pass Group SHARED (entry)
    - Recycle Bin
      - Pass Group DELETED SHARED (entry deleted from Group SHARED)

2. On group TEST -> Edit Group -> KeeShare -> Type: Import -> Path: shared.kdbx -> Password: set -> OK

Now I see everything from shared.kdbx inside TEST which is correct.

3. On group TEST -> Delete Group -> OK

4. On group Recycle Bin -> Empty Recycle Bin -> Empty

Now the information about shared.kdbx is nowhere in the database (at least in the visible fields)

5. Click on Root LOCAL -> New Group -> Name: Shared -> KeeShare -> Type: Import -> Path: shared.kdbx -> Password: set -> OK

Shared Group is created, but nothing is inside. I don't see any password or group from shared.kdbx file.

## Expected Behavior

Synchronized or imported KeeShare Group show all items from remote password database file.

## Actual Behavior

Passwords from previously deleted KeeShare kdbx files are not visible.

## Context

**Debug info:**
KeePassXC - Version 2.6.3
Revision: beae186

Qt 5.15.2
Debugging mode is disabled.

Operating system: Windows 10 Version 2009
CPU architecture: x86_64
Kernel: winnt 10.0.19042

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey

Cryptographic libraries:
- libgcrypt 1.8.7

**Comment by droidmonkey** [comment]
Good find!

**Comment by StefanIGit** [comment]
Hi, does this apply to deleting single entries too? 
I'm try to figure out how to get entries back if they have been deleted (including from recycle bin) in the DB in which they were imported.

Only (ugly) workaround create a new DB and add the shared group again.

**Comment by droidmonkey** [comment]
Yes, the lowest tracked item for deletions are entries and groups so it exactly impacts entries.

**Comment by SamuelPlentz** [comment]
Workaround:

Open the database in KeePass 2.61 and navigate to:

Tools → Database Tools → Database Maintenance → Delete Information About Deleted Objects → Delete

This removes the stored information about deleted objects and forces KeeShare to synchronize the entries again.
